cmd/argocd/commands/app.go

```diff
@@ -2331,6 +2393,11 @@ type resourceState struct {
 	Message   string
 }
 
+type versionedResourceState struct {
+	resourceState
+	Version string
+}
```

Comment:
Why not just update resourceState with an extra field?

Reply:
I did not want to introduce unintended changes to methods which do not require the field. Looking more closer at the methods which are using the `resourceState` struct I see that the change would not be breaking. I would be open to simplifying the code at that place.

Reply:
Yeah, let's simplify the code and re-use `resourceState`.

Reply:
Done.